src/backend/commands/vacuum.c

```diff
@@ -2645,15 +2645,18 @@ vacuum_appendonly_indexes(Relation aoRelation, VacuumStmt *vacstmt, Bitmapset *d
 		{
 			for (i = 0; i < nindexes; i++)
 			{
-				scan_index(Irel[i], true, elevel);
+				scan_index(Irel[i],
```

Comment:
interesting, that in 6.20.1_arenadata33 this logic is differrent
```c
		if (vacuum_appendonly_index_should_vacuum(aoRelation, vacstmt,
					&vacuumIndexState, &rel_tuple_count))
		{
			Assert(rel_tuple_count > -1.0);

			for (i = 0; i < nindexes; i++)
			{
				vacuum_appendonly_index(Irel[i], &vacuumIndexState,
										rel_tuple_count,
										elevel);
			}
			reindex_count++;
		}
		else
		{
			for (i = 0; i < nindexes; i++)
				scan_index(Irel[i], rel_tuple_count, true, elevel);
		}
```

Reply:
Take a look to mentioned in the description patch. This patch is part of 6.22.0.

Reply:
I see, I just notice that like functionality was and added again